main.nf

```diff
@@ -0,0 +1,708 @@
+
+    script:
+    if (params.singleEnd) {
+        """
+        fusioncatcher \\
+            -d ${data_dir} \\
+            -i ${reads[0]} \\
+            --threads ${task.cpus} \\
+            -o . \\
+            --skip-blat \\
+            --single-end \\
+            ${params.fc_extra_options}
+        """
+    } else {
+        """
+        fusioncatcher \\
+            -d ${data_dir} \\
+            -i ${reads[0]},${reads[1]} \\
+            --threads ${task.cpus} \\
+            -o . \\
+            --skip-blat \\
+            ${params.fc_extra_options}
+        """
+    }
+}
+
+/*
+ * Ericscript
+ */
+process ericscript {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Ericscript", mode: 'copy'
+
+    when:
+    params.ericscript || (params.ericscript && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_ericscript
+    file reference from ericscript_ref.collect()
+
+    output:
+    file './tmp/fusions.results.total.tsv' into ericscript_fusions
+    file '*' into ericscript_output
+
+    script:
+    if (!params.singleEnd) {
+        """
+        ericscript.pl \\
+            -db ${reference} \\
+            -name fusions \\
+            -p ${task.cpus} \\
+            -o ./tmp \\
+            ${reads[0]} \\
+            ${reads[1]}
+        """
+    }
+}
+
+/*
+ * Pizzly
+ */
+process pizzly {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Pizzly", mode: 'copy'
+
+    when:
+    params.pizzly || (params.pizzly && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_pizzly
+    file fasta from pizzly_fasta.collect()
+    file gtf from pizzly_gtf.collect()
+    
+    output:
+    file '*.unfiltered.json' into pizzly_fusions
+    file '*.{json,txt,tsv,fasta}' into pizzly_output
+
+    script:
+    if (!params.singleEnd) {
+        """
+        kallisto index -i index.idx -k ${params.pizzly_k} ${fasta}
+        kallisto quant -t ${task.cpus} -i index.idx --fusion -o output ${reads[0]} ${reads[1]}
+        pizzly -k ${params.pizzly_k} \\
+            --gtf ${gtf} \\
+            --cache index.cache.txt \\
+            --align-score 2 \\
+            --insert-size 400 \\
+            --fasta ${fasta} \\
+            --output pizzly_fusions output/fusion.txt
+        """
+    }
+}
+
+/*
+ * Squid
+ */
+process squid {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Squid", mode: 'copy'
+
+    when:
+    params.squid || (params.squid && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_squid
+    file index from star_index.collect()
+    file gtf
+    
+    output:
+    file '*_annotated.txt' into squid_fusions
+    file '*' into squid_output
+
+    script:
+    def avail_mem = task.memory ? "--limitBAMsortRAM ${task.memory.toBytes() - 100000000}" : ''
+    """
+    STAR \\
+        --genomeDir ${index} \\
+        --sjdbGTFfile ${gtf} \\
+        --runThreadN ${task.cpus} \\
+        --readFilesIn ${reads[0]} ${reads[1]} \\
```

Comment:
No need to specify single or double end?

Reply:
Fixed!

Reply:
..doesn't look fixed?

Reply:
Squid doesn't run because the condition will be false in `when` [clause here](https://github.com/nf-core/rnafusion/blob/fed5fd885cdc361556bdd89c7863be0d865e12f9/main.nf#L405)